modules/imgproc/src/contours_new.cpp

```diff
@@ -271,59 +277,60 @@ static void icvFetchContourEx(Mat &image,
-    int approx_method2;         /* final approx method */
+    Point offset; /* ROI offset: coordinates, added to each contour point */
+    Point pt; /* current scanner position */
+    Point lnbd; /* position of the last met contour */
+    schar nbd; /* current mark val */
+    int approx_method1; /* approx method when tracing */
+    int approx_method2; /* final approx method */
     int mode;
     CTree tree;
     array<int, 128> ctable;
 
 public:
+    ContourScanner_() {}
+    ~ContourScanner_() {}
+    inline bool isInt() const
+    {
+        return (this->mode == CV_RETR_FLOODFILL);
+    }
+    inline bool isSimple() const
+    {
+        return (this->mode == RETR_EXTERNAL || this->mode == RETR_LIST);
+    }
 
-    ContourScanner_() { }
-    ~ContourScanner_() { }
-    inline bool isInt() const { return (this->mode == CV_RETR_FLOODFILL); }
-    inline bool isSimple() const { return (this->mode == RETR_EXTERNAL || this->mode == RETR_LIST); }
-
-    CNode & makeContour(schar &nbd_, const bool is_hole, const int x, const int y);
-    bool contourScan(const int prev, int &p, Point &last_pos, const int x, const int y);
-    int findFirstBoundingContour(const Point &last_pos, const int y, const int lval, int par);
-    int findNextX(int x, int y, int & prev, int & p);
+    CNode& makeContour(schar& nbd_, const bool is_hole, const int x, const int y);
+    bool contourScan(const int prev, int& p, Point& last_pos, const int x, const int y);
+    int findFirstBoundingContour(const Point& last_pos, const int y, const int lval, int par);
+    int findNextX(int x, int y, int& prev, int& p);
     bool findNext();
 
-    static shared_ptr<ContourScanner_> create(Mat img, int mode, int  method, Point offset);
+    static shared_ptr<ContourScanner_> create(Mat img, int mode, int method, Point offset);
 }; // class ContourScanner_
 
 typedef shared_ptr<ContourScanner_> ContourScanner;
 
-
-shared_ptr<ContourScanner_> ContourScanner_::create(Mat img, int mode, int  method, Point offset)
+shared_ptr<ContourScanner_> ContourScanner_::create(Mat img, int mode, int method, Point offset)
 {
     if (mode == RETR_CCOMP && img.type() == CV_32SC1)
         mode = RETR_FLOODFILL;
 
     if (mode == RETR_FLOODFILL)
         CV_CheckTypeEQ(img.type(), CV_32SC1, "RETR_FLOODFILL mode supports only CV_32SC1 images");
     else
-        CV_CheckTypeEQ(img.type(), CV_8UC1, "Modes other than RETR_FLOODFILL and RETR_CCOMP support only CV_8UC1 images");
+        CV_CheckTypeEQ(
+            img.type(),
+            CV_8UC1,
+            "Modes other than RETR_FLOODFILL and RETR_CCOMP support only CV_8UC1 images");
 
     CV_Check(mode,
-             mode == RETR_EXTERNAL
-             || mode == RETR_LIST
-             || mode == RETR_CCOMP
-             || mode == RETR_TREE
-             || mode == RETR_FLOODFILL,
+             mode == RETR_EXTERNAL || mode == RETR_LIST || mode == RETR_CCOMP || mode == RETR_TREE
+                 || mode == RETR_FLOODFILL,
```

Comment:
less readable

Reply:
I agree. Unfortunately this is due to some weird clang-format behavior - it doesn't align arguments in a function call as in other places and operator indentation doesn't work too.

It can be fixed in following ways:
1. Add comments after each each piece, then it will only add a single offset (`ContinuationIndentWidth`) - not very good, because it incentivise addition of meaningless comments
2. Move this condition to separate variable - it would be good in general, but not in this case, because of `CV_Check` specifics - we want full condition to be visible in logs
3. Mark `CV_Check` macro as whitespace-sensitive (`WhitespaceSensitiveMacros` option), then it won't touch its internals at all - not good in general as it disables auto-formatting

I decided in this case to leave this specific check call less readable, and only partially fix the next check call, because comment is appropriate there:
```.cpp
    CV_Check(mode,
             mode == RETR_EXTERNAL || mode == RETR_LIST || mode == RETR_CCOMP || mode == RETR_TREE
                 || mode == RETR_FLOODFILL,
             "Wrong extraction mode");

    CV_Check(method,
             method == 0  // raw chain code - no enum value yet
                 || method == CHAIN_APPROX_NONE || method == CHAIN_APPROX_SIMPLE
                 || method == CHAIN_APPROX_TC89_L1 || method == CHAIN_APPROX_TC89_KCOS,
             "Wrong approximation method");
```

Perhaps it would be good to add special macro `CV_CheckIsOneOf(method, 0, CHAIN_APPROX_NONE, ..., "some message")` or some other convenient tool for similar cases.